sstables/index_reader.hh

```diff
@@ -765,7 +765,9 @@ public:
         , _use_caching(caching)
         , _single_page_read(single_partition_read) // all entries for a given partition are within a single page
     {
-        sstlog.trace("index {}: index_reader for {}", fmt::ptr(this), _sstable->get_filename());
+        if (sstlog.is_enabled(logging::log_level::trace)) {
+            sstlog.trace("index {}: index_reader for {}", fmt::ptr(this), _sstable->get_filename());
+        }
```

Comment:
Just FYI we also have [lazy_param](https://github.com/scylladb/seastar/blob/67c7df07b6d4647b51977ec46533a5043851559a/include/seastar/util/lazy.hh#L76) for situations like this, although the if is completely fine also.